pkg/services/ngalert/state/state_tracker.go

```diff
@@ -53,22 +54,36 @@ func NewStateTracker(logger log.Logger) *StateTracker {
 	return tracker
 }
 
-func (st *StateTracker) getOrCreate(uid string, orgId int64, result eval.Result) AlertState {
+func (st *StateTracker) getOrCreate(alertRule *ngModels.AlertRule, result eval.Result) AlertState {
 	st.stateCache.mu.Lock()
 	defer st.stateCache.mu.Unlock()
+	lbs := data.Labels{}
```

Comment:
I got the impression (but not completely sure) that we want to include into the alert instance labels also the rule labels. In that case this should become:
```suggestion
	lbs := alertRule.Labels
```

Reply:
We actually need to merge the labels from the rule definition + the ones from the result.

Reply:
So, to be clear, the rule.Labels will have just the labels from the AlertRule and alert.Labels will be the labels from the evaluation result AND the labels from the AlertRule. Is that correct @gotjosh @papagian ?

Reply:
I need to look a bit more in detail on how that works and what other things we might want to add here - I'll park this conversation for now but I'll get back to you on that one.